src/replaceAroundStep.ts

```diff
@@ -25,6 +28,83 @@ export function suggestReplaceAroundStep(
   prevSteps: Step[],
   suggestionId: number,
 ) {
+  // This detects and handles changes from `setNodeMarkup`
+  // (https://github.com/handlewithcarecollective/prosemirror-suggest-changes/issues/7)
+  if (
+    step.insert === 1 &&
+    step.from === 0 &&
+    step.slice.size === 2 &&
+    step.gapTo === step.to - 1 &&
+    step.gapFrom === step.from + 1 &&
+    (step as any).structure === true
+  ) {
+    const { modification } = state.schema.marks;
+    if (!modification) {
+      throw new Error(
+        `Failed to apply modifications to node: schema does not contain modification mark. Did you forget to add it?`,
+      );
+    }
+
+    const newNode = step.slice.content.firstChild!;
+    const oldNode = doc.resolve(step.from).nodeAfter!;
+
+    // handle node type changes
+    if (newNode.type.name !== oldNode.type.name) {
+      // type changed
+
+      // Code below is similar to trackAttrStep()
+      const rebasedPos = rebasePos(
+        step.from,
+        prevSteps,
+        trackedTransaction.steps,
+      );
+      const $pos = trackedTransaction.doc.resolve(rebasedPos);
+      const node = $pos.nodeAfter;
```

Comment:
Rather than using `?.` later, we should probably throw if `node` is null here, since that would cause errors if we tried to revert later/means that something went quite wrong